python/ray/serve/policy.py

```diff
@@ -47,9 +47,24 @@ class RandomEndpointPolicy(EndpointPolicy):
     be made deterministically based on the hash of the shard key.
     """
 
-    def __init__(self, traffic_dict):
+    def __init__(self, traffic_policy):
         self.backend_names, self.backend_weights = zip(
-            *sorted(traffic_dict.items()))
+            *sorted(traffic_policy.traffic_dict.items()))
+        self.shadow_backends = traffic_policy.shadow_dict
+
+    def _select_backends(self, val):
+        curr_sum = 0
+        for i in range(len(self.backend_weights)):
+            curr_sum += self.backend_weights[i]
+            if curr_sum > val:
+                chosen_backend = self.backend_names[i]
```

Comment:
need a break here otherwise it's always going to be the last one

Reply:
oops, good catch.